core/platform/PlatformMacros.h

```diff
@@ -94,6 +94,16 @@ Copyright (c) 2021 Bytedance Inc.
 #    define AX_ENABLE_CACHE_TEXTURE_DATA 0
 #endif
 
+/** @def AX_ENABLE_RESTART_APPLICATION_ON_CONTEXT_LOST
+ * Enable this if application should be restarted after the OpenGL context is lost
+ *
+ */
+#if (AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID && !AX_ENABLE_CACHE_TEXTURE_DATA)
```

Comment:
Can we make restart on by default?
Can we remove AX_ENABLE_CACHE_TEXTURE_DATA and code related to it in next PR?